qualang_tools/bakery/readme.md

```diff
@@ -0,0 +1,273 @@
+    b.add_Op("Op1", "qe1", [const_Op, const_Op2]) #qe1 is a mixInputs element
+    Op3 = [1., 1., 1.]
+    Op4 = [2., 2., 2.]
+    b.add_Op("Op2", "qe1", [Op3, Op4])
+    b.play("Op1", "qe1")   
+    # The baked waveform is at this point I: [0.3, 0.3, 0.3, 0.3, 0.3]
+    #                                     Q: [0.2, 0.2, 0.2, 0.3, 0.4]
+    b.play_at("Op3", "qe1", t=-2) #t indicates the time index where this new sample should be added
+    # The baked waveform is now I: [0.3, 0.3, 0.3, 1.3, 1.3, 1.0]
+    #                           Q: [0.2, 0.2, 0.2, 2.3, 2.4, 2.0]
+    
+At the baking exit, the config will have an updated sample 
+adapted for QUA compilation, according to the padding_method chosen, in this case: """
+I: [0, 0, 0, 0, 0, 0.3, 0.3, 0.3, 1.3, 1.3, 1.0, 0, 0, 0, 0, 0], 
+Q:  [0, 0, 0, 0, 0, 0.2, 0.2, 0.2, 2.3, 2.4, 2.0, 0, 0, 0, 0, 0]
+```
+The ``play_at()`` command can also be used as a single play statement involving a wait time and a play statement. In fact, if the time index indicated in the function is actually out of the range of the existing sample, a wait command is automatically added until reaching this time index (recall that the index corresponds to the time in ns) and starts inserting the operation indicated at this time. See the example below: 
+
+```
+with baking(config=config, padding_method="symmetric_r") as b:
+    const_Op = [0.3, 0.3, 0.3, 0.3, 0.3]
+    const_Op2 = [0.2, 0.2, 0.2, 0.3, 0.4]
+    b.add_Op("Op1", "qe1", [const_Op, const_Op2]) #qe1 is a mixInputs element
+    Op3 = [1., 1., 1.]
+    Op4 = [2., 2., 2.]
+    b.add_Op("Op2", "qe1", [Op3, Op4])
+    b.play("Op1", "qe1")   
+    # The baked waveform is at this point I: [0.3, 0.3, 0.3, 0.3, 0.3]
+    #                                     Q: [0.2, 0.2, 0.2, 0.3, 0.4]
+    b.play_at("Op3", "qe1", t=8) #t indicates the time index where this new sample should be added
+    # The baked waveform is now 
+    # I: [0.3, 0.3, 0.3, 0.3, 0.3, 0, 0, 0, 1.0, 1.0, 1.0], 
+    # Q: [0.2, 0.2, 0.2, 0.3, 0.4, 0, 0, 0, 2.0, 2.0, 2.0]}
+    #                                    
+    
+At the baking exit, the config will have an updated sample 
+adapted for QUA compilation, according to the padding_method chosen, in this case:
+I: [0.3, 0.3, 0.3, 0.3, 0.3, 0, 0, 0, 1.0, 1.0, 1.0, 0, 0, 0, 0, 0], 
+Q: [0.2, 0.2, 0.2, 0.3, 0.4, 0, 0, 0, 2.0, 2.0, 2.0, 0, 0, 0, 0, 0]
+```
+
+# The negative wait
+
+Negative wait is at the moment, just an equivalent way of writing the ``play_at()`` statement.
+
+The idea is to move backwards the time index at which the following play statement should start (wait[-3] means that the following waveform will be added on top of the existing sequence on the 3 last samples and will append the rest like a usual play statement.
+
+We have the equivalence between:
+```
+b.wait(-3)
+b.play('my_pulse',qe)
+```
+and 
+```
+b.play_at('my_pulse', qe, t=-3)
+```
+# Giving more freedom on the access of baked waveforms within QUA program

```

Comment:
Didn't we say we want to remove this part?

This would behave not as expected, both for the amplitude (if it's a matrix) and for the duration...

Reply:
I'm fine with leaving this feature, but undocumented...